wallet/src/Dscp/Wallet/Knit.hs

```diff
@@ -0,0 +1,170 @@
+      pAddress = do
+        str <- pSomeAlphaNum
+        either (fail . show) return $ addrFromText str
+
+      pCoin :: Tokenizer Coin
+      pCoin = Coin <$> P.signed (return ()) P.decimal
+
+      pSecretKey :: Tokenizer ByteString
+      pSecretKey = do
+        str <- (<>) <$> P.takeP (Just "base64") 91 <*> P.string "="
+        either (fail . show) return $ Base64.decode str
+
+instance ComponentDetokenizer Wallet where
+  componentTokenRender = \case
+    TokenAddress a -> pretty a
+    TokenCoin c -> pretty c
+    TokenSecretKey sk -> Base64.encode sk
+
+instance Elem components Wallet => ComponentLitGrammar components Wallet where
+  componentLitGrammar =
+    rule $ asum
+      [ toLit . LitAddress <$> tok (_Token . uprismElem . _TokenAddress)
+      , toLit . LitCoin <$> tok (_Token . uprismElem . _TokenCoin)
+      , toLit . LitSecretKey <$> tok (_Token . uprismElem . _TokenSecretKey)
+      ]
+
+instance ComponentPrinter Wallet where
+  componentPpLit = \case
+    LitAddress x -> text (componentTokenRender (TokenAddress x))
+    LitCoin x -> text (componentTokenRender (TokenCoin x))
+    LitSecretKey x -> text (componentTokenRender (TokenSecretKey x))
+  componentPpToken = \case
+    TokenAddress _ -> "address"
+    TokenCoin _ -> "coin"
+    TokenSecretKey _ -> "encrypted secret key"
+
+data instance ComponentCommandRepr components Wallet
+  = CommandAction (WalletFace -> IO (Value components))
+
+instance ComponentLitToValue components Wallet where
+  componentLitToValue = \case
+    LitAddress x -> ValueAddress x
+    LitCoin x -> ValueCoin x
+    LitSecretKey x -> ValueSecretKey x
+
+data instance ComponentExecContext _ _ Wallet =
+  WalletExecCtx WalletFace
+
+instance MonadIO m => ComponentCommandExec m components Wallet where
+  componentCommandExec (WalletExecCtx walletFace) (CommandAction act) =
+    liftIO $ act walletFace
+
+instance AllConstrained (Elem components) '[Wallet, Core] => ComponentCommandProcs components Wallet where
+  componentCommandProcs =
+    [
+      CommandProc
+        { cpName = "gen-key-pair"
+        , cpArgumentPrepare = identity
+        , cpArgumentConsumer = do
+            passPhrase <- getArg tyString "passphrase"
+            pure (passPhrase)
+        , cpRepr = \(passPhrase) -> CommandAction $ \WalletFace{..} -> do
+            pp <- either throwIO return . mkPassPhrase . encodeUtf8 $ passPhrase
+            (pk, sk) <- walletGenKeyPair pp
+            return . toValue . ValueList $
+              [ toValue . ValueString . pretty $ pk
+              , toValue . ValueString . pretty $ sk
+              ]
+        , cpHelp = "Generate a key pair."
+        }
+    , CommandProc
+        { cpName = "send-tx"
+        , cpArgumentPrepare = identity
+        , cpArgumentConsumer = do
+            passPhrase <- getArg tyString "passphrase"
+            secretKey <- getArg tySecretKey "secretkey"
+            address <- getArg tyAddress "address"
+            value <- getArg tyCoin "value"
```

Comment:
Btw is it possible to override argument consumer so that user can specify either `[(Address, Coin)]` or `(Address, Coin)` if the latter is easier?
We'd like to provide multi-output transactions support, although one-output transaction is still the most common use case.

Reply:
In Cardano backend it looks like this:
`send <wallet id, pass, etc> out:(tx-out <address> <value>) out:(tx-out <address> <value>) out:...`
Not sure I'll be able to make it better looking

Reply:
```
knit> send-tx "asdfasdfasdfasdf" UH1OoUs/eDxxKHMlLy09+m5Qw67i+iDVvJ6wZora1TDY6FgggZoDtW8FW7TD0iUx99vPg4q4mUydmMRZaLiT9b+sEug= (tx-out LL4qKtG4mmZF4nC8hMiTpmsG7noszDtWZhiVHuqQsEqtghWdiCFmkkMp 10) (tx-out LL4qKtG4mmZF4nC8hMiTpmsG7noszDtWZhiVHuqQsEqtghWdiCFmkkMp 20)
Not implemented: sendTx
   <passphrase>
   UH1OoUs/eDxxKHMlLy09+m5Qw67i+iDVvJ6wZora1TDY6FgggZoDtW8FW7TD0iUx99vPg4q4mUydmMRZaLiT9b+sEug=
   TxOut {txOutAddr = Address {addrHash = AbstractHash ee70896bece23911966a2a7522a8cc11c9cefb58afed3f3d13bfe65a244865aa}, txOutValue = Coin {unCoin = 10}}
   TxOut {txOutAddr = Address {addrHash = AbstractHash ee70896bece23911966a2a7522a8cc11c9cefb58afed3f3d13bfe65a244865aa}, txOutValue = Coin {unCoin = 20}}

CallStack (from HasCallStack):
  error, called at src/Universum/Debug.hs:57:11 in universum-1.1.0-EGc4GwPNR66DFTSyrseD1k:Universum.Debug
  error, called at src/Dscp/Wallet/Backend.hs:20:36 in disciplina-wallet-0.1.0.0-nMR7bvFJpEFS9IJUSXcmM:Dscp.Wallet.Backend
knit> 
```